### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by endJunction**
- Some micro-optimizations.
 - making code compile with intel compiler.
 - fixing intel compiler flags for better.

### CODE REVIEW

File: ProcessLib/BoundaryCondition/BoundaryConditionCollection.h [hunk 1]
```diff
@@ -29,7 +29,8 @@ class BoundaryConditionCollection final
     std::vector<NumLib::IndexValueVector<GlobalIndexType>> const*
     getKnownSolutions(double const t) const
     {
-        for (std::size_t i=0; i<_boundary_conditions.size(); ++i) {
+        auto const n_bcs = _boundary_conditions.size();
```
Comment (chleh) [inline ProcessLib/BoundaryCondition/BoundaryConditionCollection.h]: Does that really affect performance? n_bcs should be O(10).

Comment (endJunction) [reply ProcessLib/BoundaryCondition/BoundaryConditionCollection.h]: Good that you asked, so I checked this carefully.
if TLDR => Moving outside is wrong.

The original idea was "suggested" by some third-party compiler telling about the data dependency and this loop can not be vectorized.
Actual comparison using clang compiler shows shorter assembly and measurable run time difference for a phasefield benchmark:
```
perf stat -B -d bin/ogs_size_out_of_loop ../s/Tests/Data/PhaseField/beam3d.prj >/dev/null

 Performance counter stats for 'bin/ogs_size_out_of_loop ../s/Tests/Data/PhaseField/beam3d.prj':

       5157.698260      task-clock:u (msec)       #    1.000 CPUs utilized          
                 0      context-switches:u        #    0.000 K/sec                  
                 0      cpu-migrations:u          #    0.000 K/sec                  
            16,009      page-faults:u             #    0.003 M/sec                  
    17,588,281,335      cycles:u                  #    3.410 GHz                      (49.98%)
    36,430,071,330      instructions:u            #    2.07  insn per cycle           (62.48%)
     3,387,233,593      branches:u                #  656.734 M/sec                    (62.48%)
        33,262,852      branch-misses:u           #    0.98% of all branches          (62.48%)
    14,559,139,980      L1-dcache-loads:u         # 2822.798 M/sec                    (62.49%)
       398,547,724      L1-dcache-load-misses:u   #    2.74% of all L1-dcache hits    (62.54%)
        17,056,974      LLC-loads:u               #    3.307 M/sec                    (50.06%)
         3,496,933      LLC-load-misses:u         #   20.50% of all LL-cache hits     (50.03%)

       5.158157970 seconds time elapsed
```
and the original code:
```
perf stat -B -d bin/ogs_size_in_loop ../s/Tests/Data/PhaseField/beam3d.prj >/dev/null

 Performance counter stats for 'bin/ogs_size_in_loop ../s/Tests/Data/PhaseField/beam3d.prj':

       5046.503391      task-clock:u (msec)       #    1.000 CPUs utilized          
                 0      context-switches:u        #    0.000 K/sec                  
                 0      cpu-migrations:u          #    0.000 K/sec                  
            23,377      page-faults:u             #    0.005 M/sec                  
    17,149,369,506      cycles:u                  #    3.398 GHz                      (50.02%)
    35,034,817,459      instructions:u            #    2.04  insn per cycle           (62.54%)
     3,514,233,922      branches:u                #  696.370 M/sec                    (62.55%)
        29,960,706      branch-misses:u           #    0.85% of all branches          (62.55%)
    13,705,832,653      L1-dcache-loads:u         # 2715.907 M/sec                    (62.55%)
       408,235,481      L1-dcache-load-misses:u   #    2.98% of all L1-dcache hits    (62.50%)
        17,481,707      LLC-loads:u               #    3.464 M/sec                    (49.95%)
         3,623,419      LLC-load-misses:u         #   20.73% of all LL-cache hits     (50.01%)

       5.047226486 seconds time elapsed
```

**Comment by endJunction** [comment]
I'm checking the other micro-pessimization....

File: ProcessLib/BoundaryCondition/BoundaryConditionCollection.h (hunk 1 above)
Comment (chleh) [reply ProcessLib/BoundaryCondition/BoundaryConditionCollection.h]: I didn't intend that this degenerates into a huge runtime measurement endeavour: Only to get things straight:
E.g., `n_bcs` is three, if I set three Dirichlet BCs in the prj file, isn't it?
So this loop iterates three times per nonlinear iteration.
IMHO, therefore the runtime difference due to a code change here should be minute.
Summary: Don't measure, just merge. 😄

Comment (endJunction) [reply ProcessLib/BoundaryCondition/BoundaryConditionCollection.h]: This was also my thinking, with O(10) there would be no effect, but there is one (1% for all processes is like free lunch). I checked the other loop (in MathLib) performance too and it is less pronounced in this benchmark but still there around 0.5%.

runtime measurements are as cheap as running the code w/o the measurements.

Merging now :smiley:

**Pull request merged**